### PR TITLE
feat: updated hyperlane adapter to v3

### PR DESCRIPTION
**Pull request by sendra**
- Updated HyperlaneAdapter to V3
- Updated LayerZero to V2
- Added WormholeAdapter
- Added Deployment scripts for Celo network
- Added initial deployment for Celo network

### CODE REVIEW

File: scripts/contract_extensions/TestNetChainIds.sol [hunk 1]
```diff
@@ -16,4 +16,5 @@ library TestNetChainIds {
   uint256 constant POLYGON_ZK_EVM_GOERLI = 1442;
   uint256 constant GNOSIS_CHIADO = 10200;
   uint256 constant SCROLL_SEPOLIA = 534351;
+  uint256 constant CELO_ALFAJORES = 44787;
```
Comment (kyzia551) [inline scripts/contract_extensions/TestNetChainIds.sol:20]: Should it be also the celo adapter deployment script here?

File: src/contracts/adapters/hyperLane/HyperLaneAdapter.sol [hunk 2]
```diff
@@ -2,11 +2,11 @@
 pragma solidity ^0.8.0;
 
 import {SafeCast} from 'openzeppelin-contracts/contracts/utils/math/SafeCast.sol';
-import {IMessageRecipient} from 'hyperlane-monorepo/interfaces/IMessageRecipient.sol';
```
Comment (kyzia551) [inline src/contracts/adapters/hyperLane/HyperLaneAdapter.sol:5]: most likely we should remove `hyperlane-monorepo` from the list of git modules

File: src/contracts/adapters/hyperLane/HyperLaneAdapter.sol [hunk 3]
```diff
@@ -87,7 +82,7 @@ contract HyperLaneAdapter is BaseAdapter, IHyperLaneAdapter, IMessageRecipient {
     uint32 _origin,
     bytes32 _sender,
     bytes calldata _messageBody
-  ) external onlyMailbox {
+  ) external payable onlyMailbox {
```
Comment (kyzia551) [inline src/contracts/adapters/hyperLane/HyperLaneAdapter.sol:85]: Why is it payable now?

File: tests/adapters/HyperLaneAdapter.t.sol [hunk 4]
```diff
@@ -2,18 +2,16 @@
...
+import {HyperLaneAdapter, IHyperLaneAdapter, IMailbox, TypeCasts, StandardHookMetadata} from '../../src/contracts/adapters/hyperLane/HyperLaneAdapter.sol';
 import {ICrossChainReceiver} from '../../src/contracts/interfaces/ICrossChainReceiver.sol';
 import {IBaseAdapter} from '../../src/contracts/adapters/IBaseAdapter.sol';
-import {TypeCasts} from 'hyperlane-monorepo/libs/TypeCasts.sol';
 import {ChainIds} from '../../src/contracts/libs/ChainIds.sol';
 import {Errors} from '../../src/contracts/libs/Errors.sol';
 
 contract HyperLaneAdapterTest is Test {
   address public constant ORIGIN_FORWARDER = address(123);
   address public constant CROSS_CHAIN_CONTROLLER = address(1234);
```
Comment (kyzia551) [inline tests/adapters/HyperLaneAdapter.t.sol]: do fuzz of all this addresses

**Review by kyzia551 — CHANGES_REQUESTED** [review]

File: scripts/contract_extensions/TestNetChainIds.sol (hunk 1 above)
Comment (sendra) [reply scripts/contract_extensions/TestNetChainIds.sol:20]: for now i only added the network to have it, as i am basing all other ones on this pr. (This way on lz we can already add the path)

File: tests/adapters/HyperLaneAdapter.t.sol (hunk 4 above)
Comment (sendra) [reply tests/adapters/HyperLaneAdapter.t.sol]: added fuzz to all tests

**Review by kyzia551 — APPROVED** [review]

**Pull request merged**